contracts/core/RebalancingSetToken.sol

```diff
@@ -376,4 +389,67 @@ contract RebalancingSetToken is
         // Use inherited transferFrom function
         return super.transferFrom(_from, _to, _value);
     }
+
+    /* ============ Internal Functions ============ */
+    function parseUnitArrays()
+        internal
+    {
+        // Create interfaces for interacting with sets
+        ISetToken currentSetInterface = ISetToken(currentSet);
+        ISetToken rebalancingSetInterface = ISetToken(rebalancingSet);
+
+        // Create combined token Array
+        address[] memory oldComponents = currentSetInterface.getComponents();
+        address[] memory newComponents = rebalancingSetInterface.getComponents();
+        combinedTokenArray = oldComponents.union(newComponents);
+
+        // Get naturalUnit of both sets
+        uint256 currentSetNaturalUnit = currentSetInterface.naturalUnit();
+        uint256 rebalancingSetNaturalUnit = rebalancingSetInterface.naturalUnit();
+
+        // Get units arrays for both sets
+        uint256[] memory currentSetUnits = currentSetInterface.getUnits();
+        uint256[] memory rebalancingSetUnits = rebalancingSetInterface.getUnits();
+
+        for (uint16 i=0; i < combinedTokenArray.length; i++) {
+            // Check if component in arrays and get index if it is
+            (uint256 indexCurrent, bool isInCurrent) = oldComponents.indexOf(combinedTokenArray[i]);
+            (uint256 indexRebalance, bool isInRebalance) = newComponents.indexOf(combinedTokenArray[i]);
+
+            // Compute and push unit amounts of token in currentSet, push 0 if not in set
+            if (isInCurrent) {
+                combinedCurrentUnits.push(
+                    computeUnits(currentSetUnits[indexCurrent], currentSetNaturalUnit)
+                );
+            } else {
+                combinedCurrentUnits.push(uint256(0));
+            }
+
+            // Compute and push unit amounts of token in rebalancingSet, push 0 if not in set
+            if (isInRebalance) {
+                combinedRebalanceUnits.push(
+                    computeUnits(rebalancingSetUnits[indexRebalance], rebalancingSetNaturalUnit)
+                );
+            } else {
+                combinedRebalanceUnits.push(uint256(0));
+            }
+        }
+    }
+
+    /**
+     * Function to calculate the transfer value of a component given quantity of Set
+     *
+     * @param _unit             The units of the component token
+     * @param _naturalUnit      The natural unit of the Set token
+     */
+    function computeUnits(
```

Comment:
Do we have one of these functions elsewhere? If so, maybe we can avoid duplication by using a lib

Reply:
We have something similar in Core but don't see a need to have us make a call to it in order to calculate this.